imexport/export.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+
+	"github.com/cozy/cozy-stack/client"
+	"github.com/cozy/cozy-stack/client/auth"
+	"github.com/cozy/cozy-stack/client/request"
+)
+
+func export(tw *tar.Writer, opts *request.Options, authReq *auth.Request) error {
+	cClient := &client.Client{
+		Domain: authReq.Domain,
+		Scheme: opts.Scheme,
+		Client: opts.Client,
+
+		AuthClient: authReq.ClientParams,
+		AuthScopes: authReq.Scopes,
+		Authorizer: opts.Authorizer,
+		UserAgent:  opts.UserAgent,
+	}
+
+	root := "/Documents"
+
+	if err := cClient.WalkByPath(root, func(path string, doc *client.DirOrFile, err error) error {
+		fmt.Printf("Visited: %s  type: %s\n", path, doc.Attrs.Type)
+
+		if doc.Attrs.Type == client.DirType {
+			fmt.Println("directory")
+		} else if doc.Attrs.Type == client.FileType {
+			readCloser, err := cClient.DownloadByPath(path)
+			if err != nil {
+				return err
+			}
```

Comment:
Also, you could export this part in a separate method `func writeFile(cClient *client.Client, tw tar.Writer, doc *vfs.DirOrFileDoc)`, in order for the defered close to be called at the end of the copy instead of at the end of the copy of all files.